app/src/behaviors/behavior_hold_tap.c

```diff
@@ -472,6 +501,8 @@ static void decide_hold_tap(struct active_hold_tap *hold_tap,
             status_str(hold_tap->status), flavor_str(hold_tap->config->flavor),
             decision_moment_str(decision_moment));
     undecided_hold_tap = NULL;
+
+    release_hold_binding(hold_tap, decision_moment);
```

Comment:
Wouldn't it be better to put the if guard `if (hold_tap->config->hold_while_undecided && !hold_tap->hold_released)` from inside the function around every call of the function instead? I guess compiler could optimize it in theory?